landscape/client/monitor/ubuntuproinfo.py

```diff
@@ -28,7 +29,25 @@ def get_data(self):
         return json.dumps(ubuntu_pro_info, separators=(",", ":"))
 
 
-def get_ubuntu_pro_info():
+def get_ubuntu_pro_info() -> dict:
+    """Query ua tools for Ubuntu Pro status as JSON, parsing it to a dict.
+
+    If we are running on Ubuntu Core, Pro does not exist - returns a message
+    indicating this.
+    """
+    if IS_CORE:
+        return {
+            "errors": [
+                {
```

Comment:
There's some repetition here between the if and else conditions, could create a function and pass in the error message / parameters, and it would return the json

Reply:
I'll revisit this.